src/psyclone/psyir/nodes/omp_directives.py

```diff
@@ -2199,6 +2199,22 @@ def __init__(self, **kwargs):
         self.addchild(OMPDefaultClause(
             clause_type=OMPDefaultClause.DefaultClauseTypes.SHARED))
 
+    @staticmethod
+    def create(children=None, **kwargs):
```

Comment:
Can we make this a class method? There is the `OMPTeamDistributeParallelDoDirective` that inherits from (only) `OMPParallelDoDirective` and this implementation will repeat this issue for that class. 
That class is identical to this directive apart from _directive_string so we shouldn't have the same issue.

Reply:
Note that this reimplements `OMPParallelDirective.create` static method by a class method in the derived subclass which is maybe slightly unpythonic. Should I make the parent one a class method too?

Reply:
Yes - it doesn't look like that should have any other side effects so thats probably best.